### ISSUE TITLE
Bug: [no-empty-object-type] class/interface merge is not detected for generic interfaces

### ISSUE THREAD

**Issue by zaewc** [typescript-eslint/typescript-eslint#12539]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Playground Link

https://typescript-eslint.io/play/#ts=6.0.3&fileType=.tsx&code=MYGwhgzhAEBiD28A8AVAfNA3tAvgKAEsA7AFwFMAnAMzGDLkVQ2xyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYkACAXAngA4oDG0AlobgLQrzkB2uA9AwPa0C2V%2B1bARgCtEJGgWLooiaNDbRIAXxAKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class Foo<T> { }
interface Foo<T> { }
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

no suggestion, like the non-generic `class Foo {}` / `interface Foo {}` case

### Actual Result

It suggests `type Foo<T, U> = object` / `type Foo<T, U> = unknown`. 

Applying either leaves the class in place and creates a duplicate `Foo` identifier.

**Comment by bradzacher** [comment]
duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/1614
